### PR TITLE
Design Rework #3 - Functional Design

### LINKED ISSUE #138: Highlighting of the bottom navigation bar
When switching between fragments the bottom navigation bar is highlighted incorrectly e.g. when you navigate with the back button in the library and the search view.

### PR DESCRIPTION
**Pull request by LuisMossburger**
**Description**
See also #100 and #118 for progress. Closes #138.
This merges reworked menus & buttons, as well as list items. Summary:
* Floating button now properly floats (even if keyboard is expanded)
* The logo in the header is now a link to home fragment (as suggested in an informal usability test)
* (De)Selecting list items now works with short click after the first initial long click in all list views
* Fix bug where highlighting in bottomNavigation was not properly updated.

Next up: Toasts & Help, Home Fragment, "design sugar".

**Affects**
List views & layouts.

**Notes for Reviewer**
Please review & comment/approve. Thanks!

### CODE REVIEW

File: BibBuddy/app/src/main/java/de/bibbuddy/AuthorRecyclerViewAdapter.java [hunk 1]
```diff
@@ -96,6 +103,24 @@ public int getItemCount() {
...
+    List<AuthorItem> selectedItems = new ArrayList<>();
+    if (parent != null) {
+      int itemNumber = parent.getChildCount();
+      for (int i = 0; i < itemNumber; i++) {
+        if (parent.getChildAt(i).isSelected()) {
+          selectedItems.add(authorItemList.get(i));
+        }
+      }
+    }
+    return selectedItems;
```
Comment (claudia3) [inline BibBuddy/app/src/main/java/de/bibbuddy/AuthorRecyclerViewAdapter.java:121]: Please add some newlines for better readability. Also you don't need a local variable for itemNumber.
```suggestion
    List<AuthorItem> selectedItems = new ArrayList<>();
    
    if (parent != null) {
      for (int i = 0; i < parent.getChildCount(); i++) {
        if (parent.getChildAt(i).isSelected()) {
          selectedItems.add(authorItemList.get(i));
        }
      }
    }
    
    return selectedItems;
```

File: BibBuddy/app/src/main/java/de/bibbuddy/BookRecyclerViewAdapter.java [hunk 2]
```diff
@@ -94,6 +102,23 @@ public int getItemCount() {
...
+    List<BookItem> selectedItems = new ArrayList<>();
+    if (parent != null) {
+      int itemNumber = parent.getChildCount();
+      for (int i = 0; i < itemNumber; i++) {
+        if (parent.getChildAt(i).isSelected()) {
+          selectedItems.add(bookList.get(i));
+        }
+      }
+    }
+    return selectedItems;
```
Comment (claudia3) [inline BibBuddy/app/src/main/java/de/bibbuddy/BookRecyclerViewAdapter.java:120]: Same suggestions as in the AuthorRecyclerViewAdapter.
```suggestion
    List<BookItem> selectedItems = new ArrayList<>();
    
    if (parent != null) {
      for (int i = 0; i < parent.getChildCount(); i++) {
        if (parent.getChildAt(i).isSelected()) {
          selectedItems.add(bookList.get(i));
        }
      }
    }
    
    return selectedItems;
```

File: BibBuddy/app/src/main/java/de/bibbuddy/LibraryRecyclerViewAdapter.java [hunk 3]
```diff
@@ -111,6 +119,24 @@ public void setLibraryList(List<ShelfItem> libraryList) {
...
+    List<LibraryItem> selectedItems = new ArrayList<>();
+    if (parent != null) {
+      int itemNumber = parent.getChildCount();
+      for (int i = 0; i < itemNumber; i++) {
+        if (parent.getChildAt(i).isSelected()) {
+          selectedItems.add(libraryList.get(i));
+        }
+      }
+    }
+    return selectedItems;
```
Comment (claudia3) [inline BibBuddy/app/src/main/java/de/bibbuddy/LibraryRecyclerViewAdapter.java:137]: Same suggestions as in the other adapters.
```suggestion
    List<LibraryItem> selectedItems = new ArrayList<>();
    
    if (parent != null) {
      for (int i = 0; i < parent.getChildCount(); i++) {
        if (parent.getChildAt(i).isSelected()) {
          selectedItems.add(libraryList.get(i));
        }
      }
    }
    
    return selectedItems;
```

File: BibBuddy/app/src/main/java/de/bibbuddy/MainActivity.java [hunk 4]
```diff
@@ -23,6 +23,7 @@
   BottomNavigationView bottomNavigationView;
   FragmentManager fragmentManager;
   DatabaseHelper dbHelper;
+  private ImageButton logoButton;
```
Comment (claudia3) [inline BibBuddy/app/src/main/java/de/bibbuddy/MainActivity.java:26]: Please add a newline for better readability.
```suggestion

  private ImageButton logoButton;
```

**Review by claudia3 — CHANGES_REQUESTED** [review]
Please have a look at my suggestions. 
Maybe you could also add the updateNavigationFragment method in the BookNotesView so that the navigation bar gets highlighted when a user opens a book in the search.
What I also just realized is that when you delete multiple notes the items are still selected.

File: BibBuddy/app/src/main/java/de/bibbuddy/AuthorRecyclerViewAdapter.java (hunk 1 above)
Comment (LuisMossburger) [reply BibBuddy/app/src/main/java/de/bibbuddy/AuthorRecyclerViewAdapter.java:121]: Thanks for the suggestion, done!

File: BibBuddy/app/src/main/java/de/bibbuddy/LibraryRecyclerViewAdapter.java (hunk 3 above)
Comment (LuisMossburger) [reply BibBuddy/app/src/main/java/de/bibbuddy/LibraryRecyclerViewAdapter.java:137]: Thanks for the suggestion, done!

File: BibBuddy/app/src/main/java/de/bibbuddy/BookRecyclerViewAdapter.java (hunk 2 above)
Comment (LuisMossburger) [reply BibBuddy/app/src/main/java/de/bibbuddy/BookRecyclerViewAdapter.java:120]: Thanks for the suggestion, done!

File: BibBuddy/app/src/main/java/de/bibbuddy/MainActivity.java (hunk 4 above)
Comment (LuisMossburger) [reply BibBuddy/app/src/main/java/de/bibbuddy/MainActivity.java:26]: Thanks for the suggestion, done!

**Comment by LuisMossburger** [comment]
Thanks for the review - applied all suggestions and opening a book now also highlights the navigation bar correctly.
I could not reproduce your error with selected note items after deleting, but also think we should open an issue for that instead of doing this inside this PR? :)

**Comment by claudia3** [comment]
> Thanks for the review - applied all suggestions and opening a book now also highlights the navigation bar correctly.

> I could not reproduce your error with selected note items after deleting, but also think we should open an issue for that instead of doing this inside this PR? :)

Ok.

**Review by claudia3 — APPROVED** [review]

**Pull request merged**